sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/MergeRowsExec.scala

```diff
@@ -69,6 +71,14 @@ case class MergeRowsExec(
     copy(child = newChild)
   }
 
+  override lazy val metrics: Map[String, SQLMetric] = Map(
```

Comment:
Is there a pattern where Spark usually defines metrics? At least, we have to group them with other variables above. I also see metrics are frequently the first thing defined in an exec node. Let's be consistent.